libbeat/common/file/rotator.go

```diff
@@ -378,14 +378,15 @@ func (r *Rotator) rotate(reason rotateReason) error {
 }
 
 func (r *Rotator) rotateByInterval(reason rotateReason) error {
-	fi, err := os.Stat(r.filename)
+	_, err := os.Stat(r.filename)
+
 	if os.IsNotExist(err) {
 		return nil
 	} else if err != nil {
 		return errors.Wrap(err, "failed to rotate backups")
 	}
 
-	logPrefix := r.intervalRotator.LogPrefix(r.filename, fi.ModTime())
+	logPrefix := r.intervalRotator.LogPrefix(r.filename, time.Now())
```

Comment:
what about the situation where last entry was yesterday then service was down and now we're writing again?
shouldn't log say that the file we're trying to backup was created and logged into yesterday only?

Reply:
IIt is a matter of taste now.... 
What is more important on log rotation? See the exactly time that log rotated or see the last time the file was written?
Inside the log I can see dates and times..

Reply:
Check this out... Log rotated with file mod date time. The service went up and generated a file that I don't know if rotated in the past or when the service started again.it is better to know when the service started again?

![54924387-08987100-4eeb-11e9-920c-850ec3b8660d](https://user-images.githubusercontent.com/38509299/55059361-84aaca00-504d-11e9-84fd-128605b02f1c.png)

Reply:
```suggestion
	logPrefix := r.intervalRotator.LogPrefix(r.filename, fi.ModTime())
```